packages/core-js/modules/web.url-search-params.constructor.js

```diff
@@ -57,31 +58,130 @@ var split = uncurryThis(''.split);
 var stringSlice = uncurryThis(''.slice);
 
 var plus = /\+/g;
-var sequences = Array(4);
+var FALLBACK_REPLACER = '\uFFFD';
+var VALID_HEX = /^[0-9a-f]+$/i;
 
-var percentSequence = function (bytes) {
-  return sequences[bytes - 1] || (sequences[bytes - 1] = RegExp('((?:%[\\da-f]{2}){' + bytes + '})', 'gi'));
+var indexOf = uncurryThis(''.indexOf);
+var numberToString = uncurryThis(1.0.toString);
+var fromCharCode = String.fromCharCode;
+var fromCodePoint = getBuiltIn('String', 'fromCodePoint');
+var $isNaN = isNaN;
+var $parseInt = parseInt;
+
+var parseHexOctet = function (string, start) {
+  var substr = stringSlice(string, start, start + 2);
+  if (!VALID_HEX.test(substr)) return NaN;
+
+  return $parseInt(substr, 16);
 };
 
-var percentDecode = function (sequence) {
-  try {
-    return decodeURIComponent(sequence);
-  } catch (error) {
-    return sequence;
+var getLeadingOnes = function (octet) {
+  var binString = padStart(numberToString(octet, 2), 8, '0');
+  var firstZero = indexOf(binString, '0');
+  return firstZero !== -1 ? firstZero : binString.length;
+};
+
+var utf8Decode = function (octets) {
+  var len = octets.length;
+  var codePoint = null;
+
+  switch (len) {
```

Comment:
Since `len` is used only here, direct usage of `octets.length` here could be beter.